src/RockBridge.cpp

```diff
@@ -1,193 +0,0 @@
-void RockBridge::modelAdded(std::string const& modelName)
-{
-	gzmsg << "RockBridge: added entity: " << modelName << std::endl;
-	for(WorldContainer::iterator world_it = worlds.begin();
-		world_it != worlds.end(); ++world_it)
-	{
-		int environment = GROUND;
-		std::string world_name("underwater");
-		if(world_name.compare((*world_it)->GetName()) == 0)
-		{
-			environment = UNDERWATER; 
-			gzmsg << "RockBridge: underwater world detected. " << std::endl;
-		}	
-
-//        if(gazebo::physics::ModelPtr model = (*world_it)->GetModel(modelName))
-//        {
-////          gazebo::physics::BasePtr model_base = (*world_it)->GetByName(modelName);
-////		  gazebo::physics::ModelPtr model(new gazebo::physics::Model(model_base));
-
-//            gzmsg << "Model name: " << model->GetName() << std::endl;  		    
-//		    createTask(*world_it, model, environment);
-//        }
-	}
-}
-//======================================================================================
-void RockBridge::createTask(gazebo::physics::WorldPtr world, gazebo::physics::ModelPtr model, int environment)
-{
-	std::cout << " ~ " << std::endl;
-	gzmsg << "RockBridge: initializing model: "<< (model)->GetName() << std::endl;
-
-	// Create and initialize one rock component for each gazebo model
-	task = new gazebo::ModelTask();
-	tasks.push_back(task);
-	task->setGazeboModel(world, model, environment);
-	
-	// Export the component interface on CORBA to Ruby access the component
-	RTT::corba::TaskContextServer::Create( task );
-	
-	// Set up the component activityate_signal
-	RTT::Activity* activity = new RTT::Activity(
-			SCHED_OTHER,
-			RTT::os::LowestPriority,
-			0.0,
-			task->engine(),
-			"orogen_default_rock_gazebo");		
-//		RTT::extras::SlaveActivity* activity = new ModelActivity(task->engine());
-	
-	this->activities.push_back(activity);
-}
-//======================================================================================
-// Callback method triggered every update begin
-// It test conditions and implement all rock components functionalities
-void RockBridge::updateBegin(common::UpdateInfo const& info)
-{
-	for(ModelTasks::iterator it = tasks.begin(); it != tasks.end(); ++it)
-	{
-		(*it)->updateModel();
-	}
-
-//	for (Activities::iterator it = activities.begin(); it != activities.end(); ++it)
-//	{
-//		(*it)->execute();
-//	}
-}
-//======================================================================================
-void RockBridge::updateEnd()
-{
-}
-//======================================================================================
-RockBridge::RockBridge()
-{
-}
-//======================================================================================
-RockBridge::~RockBridge()
-{
-	delete task;
-	
-	// Delete pointers to activity
-	for(Activities::iterator activity_it = activities.begin(); 
-		    activity_it != activities.end(); ++activity_it)
-	{
-		delete *activity_it;
-	}
-	activities.clear();
```

Comment:
Read about lifetime of variables. http://stackoverflow.com/questions/11137516/scope-vs-lifetime-of-variable for instance.